beacon-chain/sync/service_test.go

```diff
@@ -3,30 +3,98 @@ package sync
 import (
 	"context"
 	"testing"
+    "time"
 
 	"github.com/prysmaticlabs/prysm/beacon-chain/internal"
+	initialsync "github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync"
 )
 
-func setupTestSyncService(t *testing.T) *Service {
+
+func NotSyncQuerierConfig() *QuerierConfig {
+	return &QuerierConfig{
+		ResponseBufferSize: 100,
+		CurentHeadSlot : 10,
+	}
+}
+
+func initializeTestSyncService(ctx context.Context, cfg *Config, synced bool) *Service {
+	var sqCfg *QuerierConfig
+	if synced {
+		sqCfg = DefaultQuerierConfig()
+     } else {
+		sqCfg = NotSyncQuerierConfig()
+	 }
+	sqCfg.BeaconDB = cfg.BeaconDB
+	sqCfg.P2P = cfg.P2P
+
+	isCfg := initialsync.DefaultConfig()
+	isCfg.BeaconDB = cfg.BeaconDB
+	isCfg.P2P = cfg.P2P
+
+	rsCfg := DefaultRegularSyncConfig()
+	rsCfg.ChainService = cfg.ChainService
+	rsCfg.BeaconDB = cfg.BeaconDB
+	rsCfg.P2P = cfg.P2P
+
+	sq := NewQuerierService(ctx, sqCfg)
+	rs := NewRegularSyncService(ctx, rsCfg)
+
+	isCfg.SyncService = rs
+	is := initialsync.NewInitialSyncService(ctx, isCfg)
+
+	return &Service{
+		RegularSync: rs,
+		InitialSync: is,
+		Querier:     sq,
+	}
+
+}
+
+
+func setupTestSyncService(t *testing.T, synced bool) *Service {
 	db := internal.SetupDB(t)
 	defer internal.TeardownDB(t, db)
 
-	if err := db.InitializeState(); err != nil {
+	unixTime := uint64(time.Now().Unix())
+	deposits := setupInitialDeposits(t)
+	if err := db.InitializeState(unixTime, deposits); err != nil {
 		t.Fatalf("Failed to initialize state: %v", err)
 	}
 
 	cfg := &Config{
 		ChainService:  &mockChainService{},
 		P2P:           &mockP2P{},
 		BeaconDB:      db,
-		AttestService: &mockAttestService{},
+		OperationService : &mockOperationService{},
 	}
+	return initializeTestSyncService(context.Background(), cfg, synced)
+}
+
+func TestStatusWithSyncedNetwork(t *testing.T) {
+	ss := setupTestSyncService(t, true)
+	synced, querierErr := ss.Querier.IsSynced()
 
-	return NewSyncService(context.Background(), cfg)
+	if synced {
```

Comment:
This test doesnt make sense, why do you check for if synced and if not synced in the same test ? This test will follow the same path when you run it over a 100 times so one of these conditions will never be proceeded with by the test runner. It cannot be both synced and !synced in a test; its either one of those conditions depending on how you set up the sync service.